src/network.rs

```diff
@@ -1,3 +1,39 @@
 //! Network specific utils
 
+use std::fmt::Debug;
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::mv_memory::MvMemory;
+
+/// A chain ID (u64) associated with relevant utils.
+pub trait PevmChain {
+    /// The error type for [Self::build_mv_memory].
+    type BuildMvMemoryError: Debug + Clone;
+
+    /// The error type for [Self::get_block_spec].
+    type GetBlockSpecError: Debug + Clone;
+
+    /// The error type for [Self::get_gas_price].
+    type GetGasPriceError: Debug + Clone;
+
+    /// Get chain id.
+    fn id(&self) -> u64;
+
+    /// Build [MvMemory]
+    fn build_mv_memory(
+        hasher: &ahash::RandomState,
+        block_env: &BlockEnv,
+        txs: &[TxEnv],
+    ) -> Result<MvMemory, Self::BuildMvMemoryError>;
+
+    /// Get block's [SpecId]
+    fn get_block_spec(header: &Header) -> Result<SpecId, Self::GetBlockSpecError>;
```

Comment:
`id` -> `get_block_spec` -> `get_gas_price` -> `build_mv_memory` is a better order.

Reply:
All functions should have `&self` in case the implementing `struct` stores more information to use in the function implementations. For instance, I'd imagine different Ethereum networks having different hardfork timestamps for the same `SpecId`. A proper implementation would need to match on the chain id, or hardfork timestamps provided by the struct.